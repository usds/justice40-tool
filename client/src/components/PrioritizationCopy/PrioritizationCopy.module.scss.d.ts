declare namespace PrioritizationCopyNamespace {
    export interface IPrioritizationCopyScss {
      prioCopyPara2: string;
    }
  }

declare const PrioritizationCopyScssModule: PrioritizationCopyNamespace.IPrioritizationCopyScss & {
    /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
    locals: PrioritizationCopyNamespace.IPrioritizationCopyScss;
  };

  export = PrioritizationCopyScssModule;
