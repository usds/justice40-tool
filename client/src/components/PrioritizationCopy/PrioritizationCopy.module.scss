@use '../../styles/design-system.scss' as *;

.prioCopyPara2{
    @include u-padding-top(2);
}
