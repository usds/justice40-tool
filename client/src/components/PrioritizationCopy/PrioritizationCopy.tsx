import React from 'react';

// import * as styles from './PrioritizationCopy.module.scss';

import * as EXPLORE_COPY from '../../data/copy/explore';

interface IPrioritizationCopy {
  totalCategoriesPrioritized: number
  totalBurdensPrioritized: number
  isAdjacencyThreshMet: boolean,
  isAdjacencyLowIncome: boolean,
  tribalCountAK: number | null,
  tribalCountUS: null, // when this signal is supported add number type
  percentTractTribal: number | null
};

/**
 * This component returns the prioritzation copy.
 *
 * The truth table that tracks all the states for this logic is captured in the shared doc
 * called "Indicators for UI and BE Signals". The Sheet is called "Disadv Copy v4 p1". This
 * spreadsheet is a giant truth table with all possible BE signal combinations and what the
 * approppriate copy should be.
 *
 * @param {number} totalCategoriesPrioritized
 * @param {number} totalBurdensPrioritized
 * @param {boolean} isAdjacencyThreshMet
 * @param {boolean} isAdjacencyLowIncome
 * @param {number | null} tribalCountAK
 * @param {number | null} tribalCountUS
 * @param {number | null} percentTractTribal
 * @return {JSX}
 */
const PrioritizationCopy =
   ({totalCategoriesPrioritized,
     totalBurdensPrioritized,
     isAdjacencyThreshMet,
     isAdjacencyLowIncome,
     tribalCountAK,
     tribalCountUS,
     percentTractTribal,
   }:IPrioritizationCopy) => {
     let prioCopyRendered;

     // if 1
     if (totalCategoriesPrioritized === 0) {
       // if 1-1
       if (isAdjacencyThreshMet && isAdjacencyLowIncome) {
         prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.PRIO_SURR_LI;
         // if 1-2
       } else if (isAdjacencyThreshMet && !isAdjacencyLowIncome) {
         // if 1-2-1
         if ( tribalCountAK === null && tribalCountUS === null) {
           // if 1-2-1-1
           if (percentTractTribal === null) {
             prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.NOT_PRIO_SURR_LI;
             // if 1-2-1-2
           } else if (percentTractTribal === 0) {
             prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.PAR_PRIO_SURR_NO_LI;
             // if 1-2-1-3
           } else if (percentTractTribal >= 1) {
             prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.PAR_PRIO_SURR_NO_LI;
           }
           // if 1-2-2
         } else if (tribalCountAK !== null && tribalCountAK >= 1) {
           prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.PAR_PRIO_SURR_NO_LI;
         }
         // if 1-3
       } else if (
         !(isAdjacencyThreshMet && isAdjacencyLowIncome) &&
          tribalCountAK === null &&
          tribalCountUS === null &&
          percentTractTribal === null
       ) {
         // if 1-3-1
         if (totalBurdensPrioritized === 0) {
           prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.NOT_PRIO;
           // if 1-3-2
         } else if (totalBurdensPrioritized === 1) {
           prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.NOT_PRIO_1BUR;
           // if 1-3-3
         } else if (totalBurdensPrioritized > 1) {
           prioCopyRendered = EXPLORE_COPY.PRIORITIZATION_COPY.NOT_PRIO_NBUR;
         }
         // if 1-4
       } else if (!(isAdjacencyThreshMet && isAdjacencyLowIncome)) {
         // if 1-4-1
         if (
           (tribalCountAK !== null && tribalCountAK >= 1) &&
        (tribalCountUS !== null && tribalCountUS >= 1)
         ) {
           prioCopyRendered = EXPLORE_COPY.getPrioAKUSCopy(tribalCountAK, tribalCountUS);
           // if 1-4-2
         } else if (
           (tribalCountAK !== null && tribalCountAK >= 1) &&
        tribalCountUS === null
         ) {
           prioCopyRendered = EXPLORE_COPY.getPrioANVCopy(tribalCountAK);
           // if 1-4-3
         } else if (
           (tribalCountUS !== null && tribalCountUS >= 1) &&
        tribalCountAK === null
         ) {
           if (percentTractTribal === null) {
             prioCopyRendered = EXPLORE_COPY.getPrioFRTPointsCopy(`${tribalCountUS}`);
           } else if (percentTractTribal === 0) {
             prioCopyRendered = EXPLORE_COPY.getPrioFRTCopy(`less than 1%`);
           } else if (percentTractTribal >= 1) {
             prioCopyRendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`);
           }
         // 1-4-4
         } else if (tribalCountUS === null && tribalCountAK === null) {
           if (percentTractTribal === 0) {
             prioCopyRendered = EXPLORE_COPY.getPrioFRTCopy(`less than 1%`);
           } else if (percentTractTribal !== null && percentTractTribal >= 1 ) {
             prioCopyRendered = EXPLORE_COPY.getPrioFRTCopy(`${percentTractTribal}%`);
           }
         }
       }
       // if 2
     } else if (totalCategoriesPrioritized > 0) {
       if (totalBurdensPrioritized === 0) {
         prioCopyRendered = <></>;
       } else if (totalBurdensPrioritized === 1) {
         prioCopyRendered = EXPLORE_COPY.getPrioNBurdenCopy(`1`);
       } else if (totalBurdensPrioritized > 1) {
         prioCopyRendered = EXPLORE_COPY.getPrioNBurdenCopy(`more than 1`);
       }
     } else {
       prioCopyRendered = <></>;
     };

     return (
       <div>
         {prioCopyRendered}
       </div>
     );
   };

export default PrioritizationCopy;
