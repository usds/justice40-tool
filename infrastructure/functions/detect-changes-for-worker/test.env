REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1