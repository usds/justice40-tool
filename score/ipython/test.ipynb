{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
