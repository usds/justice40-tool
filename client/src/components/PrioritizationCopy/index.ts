import PrioritizationCopy from './PrioritizationCopy';
export default PrioritizationCopy;
