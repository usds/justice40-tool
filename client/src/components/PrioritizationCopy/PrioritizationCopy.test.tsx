import React from 'react';
import {render, screen} from '@testing-library/react';
import {LocalizedComponent} from '../../test/testHelpers';
import PrioritizationCopy from './PrioritizationCopy';

describe('rendering of PrioritizationCopy Component -', () => {
  const testCases = [
    {
      totalCategoriesPrioritized: 0,
      totalBurdensPrioritized: 0,
      isAdjacencyThreshMet: false,
      isAdjacencyLowIncome: false,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: null,
      // eslint-disable-next-line max-len
      para1: `This tract is not considered disadvantaged. It does not meet any burden thresholds `,
    },
    {
      totalCategoriesPrioritized: 0,
      totalBurdensPrioritized: 1,
      isAdjacencyThreshMet: false,
      isAdjacencyLowIncome: false,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: null,
      // eslint-disable-next-line max-len
      para1: `This tract is not considered disadvantaged. It meets 1 burden threshold`,
    },
    {
      totalCategoriesPrioritized: 0,
      totalBurdensPrioritized: 5,
      isAdjacencyThreshMet: false,
      isAdjacencyLowIncome: false,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: null,
      // eslint-disable-next-line max-len
      para1: `This tract is not considered disadvantaged. It meets more than 1 burden threshold `,
    },
    {
      totalCategoriesPrioritized: 2,
      totalBurdensPrioritized: 1,
      isAdjacencyThreshMet: false,
      isAdjacencyLowIncome: false,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: null,
      // eslint-disable-next-line max-len
      para1: `This tract is considered disadvantaged because it meets 1 burden threshold `,
    },
    {
      totalCategoriesPrioritized: 2,
      totalBurdensPrioritized: 5,
      isAdjacencyThreshMet: false,
      isAdjacencyLowIncome: false,
      tribalCountAK: null,
      tribalCountUS: null,
      percentTractTribal: null,
      // eslint-disable-next-line max-len
      para1: `This tract is considered disadvantaged because it meets more than 1 burden threshold `,
    },

    // {
    //   isDonut: false, percentTribal: 0,
    //   totalCategories: 0, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `The less than 1% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: 0,
    //   totalCategories: 0, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `The less than 1% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `The less than 1% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `The less than 1% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `The less than 1% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },

    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 0, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 0, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 0, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 1, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 1, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },
    // {
    //   isDonut: false, percentTribal: .31,
    //   totalCategories: 1, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `The 31% of this tract that are Federally-Recognized Tribal lands are considered disadvantaged.`,
    // },

    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 0, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },
    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 0, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },
    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 0, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },
    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 1, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },
    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 1, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },
    // {
    //   isDonut: true, percentTribal: null,
    //   totalCategories: 1, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // },

    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 0, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 0, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 0, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: 0,
    //   totalCategories: 1, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The less than 1% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },

    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 0, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 0, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 0, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 1, totalIndicators: 0,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 1, totalIndicators: 1,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
    // {
    //   isDonut: true, percentTribal: .29,
    //   totalCategories: 2, totalIndicators: 2,
    // eslint-disable-next-line max-len
    //   para1: `This tract is considered disadvantaged because it is surrounded by tracts that are disadvantaged AND meets an adjusted low income threshold.`,
    // eslint-disable-next-line max-len
    //   para2: `The 29% of this tract that are Federally-Recognized Tribal lands are also considered disadvantaged.`,
    // },
  ];

  testCases.forEach((testCase) => {
    // eslint-disable-next-line max-len
    it(`checks if component renders ${testCase.para1} when totCats = ${testCase.totalCategoriesPrioritized}, totBurds = ${testCase.totalBurdensPrioritized}, isAdj = ${testCase.isAdjacencyThreshMet}, isAdjLI = ${testCase.isAdjacencyLowIncome}, tribal % = ${testCase.percentTractTribal},`, () => {
      const {asFragment} = render(
          <LocalizedComponent>
            <PrioritizationCopy
              totalCategoriesPrioritized={testCase.totalCategoriesPrioritized}
              totalBurdensPrioritized={testCase.totalBurdensPrioritized}
              isAdjacencyThreshMet={testCase.isAdjacencyThreshMet}
              isAdjacencyLowIncome={testCase.isAdjacencyLowIncome}
              tribalCountAK={testCase.tribalCountAK}
              tribalCountUS={null}
              percentTractTribal={testCase.percentTractTribal}

            />
          </LocalizedComponent>,
      );
      expect(asFragment()).toMatchSnapshot();

      screen.getByText((content, element) => {
        return element?.tagName.toLowerCase() === 'div' && content.startsWith(testCase.para1);
      });
    });
  });
});
