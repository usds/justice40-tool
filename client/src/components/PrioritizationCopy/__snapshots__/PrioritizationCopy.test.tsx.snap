// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of PrioritizationCopy Component - checks if component renders This tract is considered disadvantaged because it meets 1 burden threshold  when totCats = 2, totBurds = 1, isAdj = false, isAdjLI = false, tribal % = null, 1`] = `
<DocumentFragment>
  <div>
    This tract is considered disadvantaged because it meets 1 burden threshold 
    <strong>
      AND
    </strong>
     the associated socioeconomic threshold.
  </div>
</DocumentFragment>
`;

exports[`rendering of PrioritizationCopy Component - checks if component renders This tract is considered disadvantaged because it meets more than 1 burden threshold  when totCats = 2, totBurds = 5, isAdj = false, isAdjLI = false, tribal % = null, 1`] = `
<DocumentFragment>
  <div>
    This tract is considered disadvantaged because it meets more than 1 burden threshold 
    <strong>
      AND
    </strong>
     the associated socioeconomic threshold.
  </div>
</DocumentFragment>
`;

exports[`rendering of PrioritizationCopy Component - checks if component renders This tract is not considered disadvantaged. It does not meet any burden thresholds  when totCats = 0, totBurds = 0, isAdj = false, isAdjLI = false, tribal % = null, 1`] = `
<DocumentFragment>
  <div>
    This tract is not considered disadvantaged. It does not meet any burden thresholds 
    <strong>
      OR
    </strong>
     at least one associated socioeconomic threshold.
  </div>
</DocumentFragment>
`;

exports[`rendering of PrioritizationCopy Component - checks if component renders This tract is not considered disadvantaged. It meets 1 burden threshold when totCats = 0, totBurds = 1, isAdj = false, isAdjLI = false, tribal % = null, 1`] = `
<DocumentFragment>
  <div>
    This tract is not considered disadvantaged. It meets 1 burden threshold 
    <strong>
      BUT
    </strong>
     no associated socioeconomic thresholds.
  </div>
</DocumentFragment>
`;

exports[`rendering of PrioritizationCopy Component - checks if component renders This tract is not considered disadvantaged. It meets more than 1 burden threshold  when totCats = 0, totBurds = 5, isAdj = false, isAdjLI = false, tribal % = null, 1`] = `
<DocumentFragment>
  <div>
    This tract is not considered disadvantaged. It meets more than 1 burden threshold 
    <strong>
      BUT
    </strong>
     no associated socioeconomic thresholds.
  </div>
</DocumentFragment>
`;
